packages/manager/modules/core/src/index.js

```diff
@@ -158,8 +158,12 @@ export const registerCoreModule = (environment) => {
     .run(
       /* @ngInject */ ($rootScope, coreConfig) => {
         $rootScope.$on('lang.onChange', (event, { lang }) => {
-          coreConfig.setUserLocale(lang);
-          window.location.reload();
+          if (shellClient) {
```

Comment:
instead of passing a shellClient instance to core module i think it's cleaner to make the onChange callback function configurable (if not set do the default reload behavior, otherwise call the callback)

that way manager-core is not tied to ovh-shell

Reply:
This done